From: Jiri Slaby <jslaby@suse.cz>
Subject: ia64 sa_restorer fix
Patch-mainline: not yet, reported as a bug in 3.0.70

In 3.0.70, commit 42cd41128edb092f9ae6fe89dbb5456963568602 (signal:
always clear sa_restorer on execve), upstream commit
2ca39528c01a933f6689cd6505ce65bd6d68a530, added a line to zero
sa_restorer. But that one is not on some archs like IA64 or MIPS. We
care only about IA64 here, so add a simple hackish condition in there
until this gets resolved properly.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 kernel/signal.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/kernel/signal.c
+++ b/kernel/signal.c
@@ -437,9 +437,11 @@ flush_signal_handlers(struct task_struct
 		if (force_default || ka->sa.sa_handler != SIG_IGN)
 			ka->sa.sa_handler = SIG_DFL;
 		ka->sa.sa_flags = 0;
+#ifndef CONFIG_IA64
 #ifdef SA_RESTORER
 		ka->sa.sa_restorer = NULL;
 #endif
+#endif
 		sigemptyset(&ka->sa.sa_mask);
 		ka++;
 	}
